components/nimbus/src/enrollment.rs

```diff
@@ -2077,6 +2089,63 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_evolve_enrollments_error_handling() -> Result<()> {
+        let existing_enrollments = vec![ExperimentEnrollment {
+            slug: "secure-gold".to_owned(),
+            status: EnrollmentStatus::Enrolled {
+                enrollment_id: Uuid::new_v4(),
+                branch: "hello".to_owned(), // XXX this OK?
+                reason: EnrolledReason::Qualified,
+                feature_id: "some_control".to_owned(),
+            },
+        }];
+
+        let _ = env_logger::try_init();
+        let (nimbus_id, app_ctx, aru) = local_ctx();
+        let evolver = EnrollmentsEvolver::new(&nimbus_id, &aru, &app_ctx);
+
+        // test that evolve_enrollments correctly handles the case where a
+        // record without a previous enrollment gets dropped
+        let test_experiments = get_test_experiments();
+
+        // this should not return an error
+        let (enrollments, events) =
+            evolver.evolve_enrollments(true, &test_experiments, &test_experiments, &[])?;
+
+        assert_eq!(
+            enrollments.len(),
+            0,
+            "no new enrollments should have been returned"
+        );
+
+        // XXX is this right?  should we be returning some sort of event for this?
+        assert_eq!(
+            events.len(),
+            0,
+            "no new enrollments should have been returned"
```

Comment:
I'm wondering if maybe we should be returning some sort of event for this.  Thoughts?

Reply:
SDK-309 filed, and I've mentioned it in the error logging comments.